app/views/events/search.html.slim

```diff
@@ -0,0 +1,22 @@
+h4 Event's name
+= form_tag(search_path, method: :get) do
+  = text_field_tag :search, params[:search]
+  h4 Category
+  - @categories.each do |category|
+    = check_box_tag "cat[]", category, false, :cat => category
```

Comment:
`check_box_tag "cat[]", category, params.fetch(:cat, []).include?('category'), :cat => category`
to powinno zaznaczyć poprzednio wybrane checkboxy